Finished Lesson 2

arturmerkulov opened the pull request.

monstarrrstudio committed:
Create main.cpp

a-starunskyi committed:
Revert "Create main.cpp"

This reverts commit a179c0ca421da8d0923052509150e5a5f7db0746.

a-starunskyi committed:
Merge branch 'main' into lesson1

a-starunskyi committed:
Merge branch 'main' into lesson1

a-starunskyi committed:
Merge branch 'main' into lesson1

a-starunskyi committed:
Merge branch 'main' into lesson1

a-starunskyi committed:
lesson1 init

arturmerkulov committed:
Finished Lesson 1

a-starunskyi committed:
added homework

a-starunskyi committed:
added filters

arturmerkulov committed:
Merge branch 'lesson2' into a_frolov

a-starunskyi committed:
updated solution

a-starunskyi committed:
update solution settings

arturmerkulov committed:
Merge branch 'lesson2' into a_frolov

a-starunskyi committed:
added lesson4 homework

arturmerkulov committed:
lesson2

a-starunskyi committed:
update homework project

a-starunskyi committed:
add lesson5 homework

arturmerkulov committed:
Finished Lesson 2

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson3' into a_frolov

a-starunskyi commented on lesson2/task1_defining_variables/main.cpp:
```diff
@@ -5,11 +5,18 @@
-    // TODO: Assign values
+    int a0 = 100;
+    char a1 = 'a';
+    bool Boolean = true;
```
variables naming should start from small character. Also the logic of naming is broken. Why not a2?

a-starunskyi commented on lesson2/task2_using_variables/main.cpp:
```diff
@@ -7,8 +7,15 @@ using namespace std;
 
 int main() {
     // TODO: Prompt user for length and width
+    int length, width, area;
```
Variables are not initialized. Also writing variables separately more readable

a-starunskyi commented on lesson2/task3_defining_using_constants/main.cpp:
```diff
@@ -6,13 +6,12 @@
-
-    // TODO: Print area
+    const double Pi = 3.14159;
+    double radius, area
```
; is missed

a-starunskyi commented on lesson2/task3_defining_using_constants/main.cpp:
```diff
@@ -6,13 +6,12 @@
-    // TODO: Print area
+    const double Pi = 3.14159;
+    double radius, area
+        cout << "Input radius: ";
```
indent is broken

a-starunskyi commented on lesson2/task6_bonus/main.cpp:
```diff
@@ -6,11 +6,16 @@
 using namespace std;
 
 int main() {
+    constexpr double Pi() { 22.0 / 7 };
```
you cannot declare function inside another function. Here works optimization but if it does not work your code will not compile

a-starunskyi commented on lesson2/task6_bonus/main.cpp:
```diff
@@ -6,11 +6,16 @@
 
 int main() {
+    constexpr double Pi() { 22.0 / 7 };
+    int rad;
```
use more readable variable naming. In radius word not so many characters :)

arturmerkulov committed:
Finished Lesson 3

arturmerkulov committed:
Lesson 3 Fixed

a-starunskyi committed:
add lesson6 homework

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson4' into a_frolov

arturmerkulov committed:
Lesson 4 Finished

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson5' into a_frolov

arturmerkulov committed:
Lesson 4 Fixed

a-starunskyi committed:
Added lesson 7 homework

a-starunskyi committed:
init lesson 8 commit: configure the project

a-starunskyi committed:
added x64 lib

arturmerkulov committed:
LESSON 5

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson8' into a_frolov

a-starunskyi committed:
added lesson code

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson6' into a_frolov

arturmerkulov committed:
LESSON 6

arturmerkulov committed:
LESSON 5 FIXED

arturmerkulov committed:
LESSON 6 FIXED

In task 9, there was a commentary mentioning that 3 doesn't work, but I have double checked and it works. Everything else is fixed.

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson7' into a_frolov

a-starunskyi committed:
added lesson 9

arturmerkulov committed:
LESSON 7

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson8' into a_frolov

a-starunskyi committed:
changed to classes

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson9' into a_frolov

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson8' into a_frolov

arturmerkulov committed:
LESSON 7 FIXED

Task 9

a-starunskyi committed:
Implemented pipe and pipeManager classes

a-starunskyi committed:
init commit

arturmerkulov committed:
Merge remote-tracking branch 'origin/lesson-tetris' into a_frolov

arturmerkulov committed:
Tetris Render

arturmerkulov committed:
Tetris (WIP)

arturmerkulov committed:
/

arturmerkulov committed:
/
